### ISSUE TITLE
linter: eslint-plugin-import(extensions) false positives with missing extension

### ISSUE THREAD

**Issue by drvn-mr** [oxc-project/oxc#12220]
### What version of Oxlint are you using?

1.6.0

### What command did you run?

`oxlint -c .oxlintrc.json test.ts`

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "env": {
    "browser": true,
    "node": true
  },
  "plugins": [
    "import"
  ],
  "rules": {
    "import/extensions": [
      "error",
      "ignorePackages",
      {
        "js": "never",
        "ts": "never"
      }
    ]
  }
}
```

### What happened?

Import extensions are always required for non-package paths (relative, aliased) since #11872.
With ESLint and `eslint-plugin-import`, there are no errors when extension is missing.

Continuing from #11818.

#### Code
```typescript
import { Package } from 'vue'; // OK, ignorePackages
import { A } from './something.hooks'; // OK
import { B } from './something.ts'; // OK, File extension "ts" should not be included in the import declaration.

import { C } from './something'; // Error, Missing file extension in import declaration
import { D } from '~/common/something'; // Error, Missing file extension in import declaration
```

#### `oxlint` output
```console
❯ oxlint -c .oxlintrc.json test.ts

  × eslint-plugin-import(extensions): Missing file extension in import declaration
   ╭─[test.ts:4:1]
 3 │ 
 4 │ import { C } from './something'; // Error, Missing file extension in import declaration
   · ────────────────────────────────
 5 │ import { D } from '~/common/something'; // Error, Missing file extension in import declaration
   ╰────
  help: Add a file extension to this import.

  × eslint-plugin-import(extensions): File extension "ts" should not be included in the import declaration.
   ╭─[test.ts:2:1]
 1 │ import { A } from './something.hooks'; // OK
 2 │ import { B } from './something.ts'; // OK, File extension "ts" should not be included in the import declaration.
   · ───────────────────────────────────
 3 │ 
   ╰────
  help: Remove the file extension from this import.

  × eslint-plugin-import(extensions): Missing file extension in import declaration
   ╭─[test.ts:5:1]
 4 │ import { C } from './something'; // Error, Missing file extension in import declaration
 5 │ import { D } from '~/common/something'; // Error, Missing file extension in import declaration
   · ───────────────────────────────────────
   ╰────
  help: Add a file extension to this import.
```

**Comment by drvn-mr** [comment]
I don't really understand why the ESLint plugin allows these when the rule is described as such:
> If it is the string "ignorePackages", then the rule enforces the use of extensions for all import statements except package imports.

So maybe `oxlint` is doing the right thing but it's a different behaviour in our setup.

It might also be caused by aliases but it also ignores missing extensions for relative imports.
> Unfortunately, in more advanced linting setups, such as when employing custom specifier aliases (e.g. you're using eslint-import-resolver-alias, paths in tsconfig.json, etc), this rule can be too coarse-grained when determining which imports to ignore and on which to enforce the config

**Comment by drvn-mr** [comment]
I can't really seem to reproduce the issue in our project anymore.

**Comment by sgarfinkel** [comment]
I am experiencing this issue, was this bug reintroduced somehow?

**Comment by drvn-mr** [comment]
I'll reopen because it still happens occasionally.
The basics seem to be that `oxlint` always expects an extension while `ESLint` is more lax about it since it has resolving capabilities.

**Comment by sgarfinkel** [comment]
For now I’ve just setup my oxlint/eslint config to handle this rule with eslint but obviously that’s not ideal.

**Comment by theoludwig** [comment]
The option `ignorePackages` seems like to not be used.

For example, I've got the issue with the following import:
```ts
import { fn } from "storybook/test"
```

**Comment by avocadowastaken** [comment]
We want to enforce usage of extensions to follow module resolution standard, and have false positives:

```json
"import/extensions": [
  "error",
  "always",
  {
    "ignorePackages": true,
    "checkTypeImports": true
  }
],
```

```ts
import "foo"; // ok
import "foo/bar"; // fails - unexpected

import "./foo"; // fails - as expected
import "./foo.ts" // ok
```

**Comment by Akiyamka** [comment]
```ts
import { createContext } from  'preact/compat' // error - unexpected
import { signal } from "@preact/signals"; // ok - as expected
```

**Comment by sebastienbarre** [comment]
Same here, though I am not even sure I understand what the correct syntax is. I tried many variants, including this monstrosity below, but there is a bug going on, right?:
```json
"import/extensions": [
  "error",
  "ignorePackages",
  { "js": "ignorePackages", "ts": "ignorePackages", "ignorePackages": true }
 ],
```

Were [fix(linter/extensions): false positives with non configured extensions by camc314 · Pull Request #11872 · oxc-project/oxc](https://github.com/oxc-project/oxc/pull/11872) and [fix(linter): fix false positives reported when analyzing package imports by taearls · Pull Request #14602 · oxc-project/oxc](https://github.com/oxc-project/oxc/pull/14602) supposed to fix it? One of them looks merged, but it's still bugging on "import type { LibSQLDatabase } from 'drizzle-orm/libsql';" for example in today's 1.25.0.

**Comment by connorshea** [comment]
Marked this as higher priority since it seems to be a problem effecting a few folks and I seem to be seeing this issue when testing repos like mastodon as well when experimenting with oxlint on it

If anyone can provide simple reproduction repo(s) of problematic behavior that'd be incredibly helpful for testing purposes. Preferably, with both ESLint and oxlint to show the discrepancies.

**Comment by connorshea** [comment]
Potentially related to problems with this rule, including a repro case of the issue with `ignorePackages` not working correctly: https://github.com/oxc-project/oxc/issues/11881

**Comment by sebastienbarre** [comment]
Why close it when the configuration is so obscure that you can't make it work. I can't figure out if my configuration is wrong, or if the code is not working.

EDIT:  the original import/extensions rule doc definitely helps (below), thanks.

**Comment by connorshea** [comment]
You didn't really need to post this in multiple places, but I'll repost my reply regardless.

> Why close it when the configuration is so obscure that you can't make it work. I can't figure out if my configuration is wrong, or if the code is not working.

There are much more constructive ways of phrasing this :) Also the fix hasn't been released yet, it'll be released on Monday.

The configuration structure is exactly the same as the original import/extensions rule: https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/extensions.md

I'll work on improving the docs relatively soon, part of the problem is that we don't support auto-doc generation for rules that take tuples yet (see #16555).

**Comment by drvn-mr** [comment]
In our project there are still errors when the import looks like this:
```ts
// file exists as common.hook.ts
import { A } from './common.hook';
```
```
  × eslint-plugin-import(extensions): File extension "hook" should not be included in the import declaration.
   ╭─[app/javascript/shared/hooks/test.ts:1:1]
 1 │ import { A } from './common.hook';
   · ──────────────────────────────────
   ╰────
  help: Remove the file extension from this import.
```

With config
```
"import/extensions": [
  "error",
  "ignorePackages",
  {
    "js": "never",
    "ts": "never",
  }
]
```

And the error disappears if we change to `"import/extensions": ["error", "ignorePackages"]`.
So it seems unknown extensions are not ignored if there are some configured ones.

Also for some reason when testing outside the project, `oxlint` reports no errors.

**Comment by drvn-mr** [comment]
Seems better in v1.37.0, only thing erroring now seems to be relative imports with unknown extensions.
